src/main/java/org/jboss/forge/arquillian/command/algeron/AddAlgeronGitRetriever.java

```diff
@@ -0,0 +1,125 @@
+package org.jboss.forge.arquillian.command.algeron;
+
+import org.jboss.forge.addon.dependencies.builder.DependencyBuilder;
+import org.jboss.forge.addon.ui.context.UIBuilder;
+import org.jboss.forge.addon.ui.context.UIContext;
+import org.jboss.forge.addon.ui.hints.InputType;
+import org.jboss.forge.addon.ui.input.UIInput;
+import org.jboss.forge.addon.ui.metadata.UICommandMetadata;
+import org.jboss.forge.addon.ui.metadata.WithAttributes;
+import org.jboss.forge.addon.ui.util.Categories;
+import org.jboss.forge.addon.ui.util.Metadata;
+
+import javax.inject.Inject;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class AddAlgeronGitRetriever extends AbstractAlgeronRetrieverCommand {
+
+    @Inject
+    @WithAttributes(shortName = 'u', label = "Git Url", required = true)
+    private UIInput<String> url;
+
+    @Inject
+    @WithAttributes(shortName = 's', label = "Username")
+    private UIInput<String> username;
+
+    @Inject
+    @WithAttributes(shortName = 'w', label = "Password", type = InputType.SECRET)
+    private UIInput<String> password;
+
+    @Inject
+    @WithAttributes(shortName = 'h', label = "Passphrase", type = InputType.SECRET)
+    private UIInput<String> passphrase;
+
+    @Inject
+    @WithAttributes(shortName = 'k', label = "Key Location", type = InputType.FILE_PICKER)
+    private UIInput<String> key;
+
+    @Inject
+    @WithAttributes(shortName = 'o', label = "Repository")
+    private UIInput<String> repository;
+
+    @Inject
+    @WithAttributes(shortName = 'd', label = "Contracts Directory")
+    private UIInput<String> contractGitDirectory;
+
+    @Inject
+    @WithAttributes(shortName = 't', label = "Tag")
+    private UIInput<String> tag;
+
+    @Inject
+    @WithAttributes(shortName = 'b', label = "Branch")
+    private UIInput<String> branch;
+
+    @Override
+    public UICommandMetadata getMetadata(UIContext context) {
+        return Metadata.from(super.getMetadata(context), getClass())
+            .category(Categories.create("Algeron"))
+            .name("Arquillian Algeron: Add Git Retriever")
+            .description("This command registers a Git Retriever for Algeron");
+    }
+
+    @Override
+    public void initializeUI(UIBuilder builder) throws Exception {
+        builder.add(url)
+            .add(username)
+            .add(password).add(passphrase)
+            .add(key)
+            .add(repository).add(contractGitDirectory)
+            .add(tag).add(branch);
+    }
+
+    @Override
+    protected DependencyBuilder getRetrieverDependency() {
+        return DependencyBuilder.create().setGroupId("org.arquillian.universe")
+            .setArtifactId("arquillian-algeron-git-retriever").setPackaging("pom").setScopeType("test");
+    }
+
+    @Override
+    protected Map<String, String> getParameters() {
+        final Map<String, String> parameters = new LinkedHashMap<>();
+        parameters.put("provider", "git");
+        parameters.put("url", url.getValue());
```

Comment:
it is a good question, yes probably you are right, let me check if Forge provides some method for doing it.